src/components/TodoList.js

```diff
@@ -18,14 +18,38 @@ const TodoList = ({ todos, deleteTodo, completeTodo }) => {
     deleteTodo(id);
   };
 
-  const listItems = todos.map(todo => (
+  let filteredTodos;
+  switch (activeTab) {
+    case 0:
+      filteredTodos = todos.filter(
+        todo => todo.isCompleted === false && todo.isExpired === false
+      );
+      break;
+    case 1:
+      filteredTodos = todos.filter(todo => todo.isCompleted === true);
+      break;
+    case 2:
+      filteredTodos = todos.filter(
+        todo => todo.isCompleted === false && todo.isExpired === true
+      );
+      break;
+    default:
+      break;
+  }
+
+  const renderCompleteButton = todo =>
+    todo.isCompleted === true || todo.isExpired === true ? false : true;
```

Comment:
this is quite hard to reason about, lets refactor it

Reply:
Thanks for your advice with and/or chaining!